python/cudf/dataframe/numerical.py

```diff
@@ -127,16 +129,26 @@ def astype(self, dtype):
                                dtype=np.dtype(dtype))
             return col
 
-    def sort_by_values(self, ascending):
-        if self.null_count > 0:
-            raise ValueError('nulls not yet supported')
-        # Clone data buffer as the key
-        col_keys = self.replace(data=self.data.copy(),
-                                dtype=self._data.dtype)
-        # Create new array for the positions
-        inds = Buffer(cudautils.arange(len(self)))
-        col_inds = self.replace(data=inds, dtype=inds.dtype)
-        cpp_sort.apply_sort(col_keys, col_inds, ascending=ascending)
+    def sort_by_values(self, ascending=True, na_position="last"):
+        new_inds = get_sorted_inds(self, ascending, na_position)
+        col_keys = cudautils.gather(data=self.data.mem,
+                                    index=new_inds.data.mem)
+        if self.mask:
+            mask = cudf.dataframe.series.Series(self)._get_mask_as_series()\
+                .take(new_inds.data.to_gpu_array()).as_mask()
+            mask = Buffer(mask)
+            # Need to gather the bitmask based on the indices as well
+            col_keys = self.replace(data=Buffer(col_keys),
+                                    mask=mask,
+                                    null_count=self.null_count,
+                                    dtype=self.dtype)
+        else:
+            col_keys = self.replace(data=Buffer(col_keys),
+                                    null_count=self.null_count,
+                                    dtype=self.dtype)
+        col_inds = self.replace(data=new_inds.data,
```

Comment:
Instead of `new_inds.data`, new_inds should be used for a gather of the existing index's data + mask.

Reply:
This is actually handled at the Series level, woops!